src/pal/src/misc/sysinfo.cpp

```diff
@@ -31,9 +31,12 @@ Revision History:
 
 #if HAVE_SYSCONF
 // <unistd.h> already included above
-#elif HAVE_SYSCTL
+#endif
+#if HAVE_SYSCTL && !defined(__linux__)
```

Comment:
Ideally I'd like to see the change in the build system.  A quick grepping also found instances of `HAVE_SYSCTL` in GC too.

Reply:
Reason why the other work is that it is guarded by HAVE_SYSCTLBYNAME and that one does not exit on Linux so it did not get included in  #27048 and it works as expected on BSD and OSX. Using it here also works if that is better than adding extra linux check. 

Should we set HAVE_SYSCTL=0 on Linux @omajid? 
Even if glibc deprecates it, musl libc may not but I'm not sure if we have any Linux functionality depending on it. From the search it seems like we use sysconf for all we need on Linux. 

BTW hw.l3cachesize works on OSX but FreeBSD does not have it. So the code would compile  but we would end up with cacheSize = 0;

Reply:
From: https://github.com/dotnet/coreclr/pull/27827

>>  I'm wondering if we should also guard rest of the code with HAVE_SYSCTLBYNAME instead of HAVE_SYSCTL so they are in balance.
>
> I think it makes sense as in this context, `HAVE_SYSCTLBYNAME` is more reliable macro to determine whether sys/sysctl.h should be included. It is also used in another file where `sysctlnametomib` is used.